--- a/src/network/ssl/qsslsocket.cpp
+++ b/src/network/ssl/qsslsocket.cpp
@@ -2963,6 +2963,7 @@ QList<QByteArray> QSslSocketPrivate::uni
         return QByteArray::fromRawData(std::begin(cstr), std::size(cstr) - 1);
     };
     static const QByteArray dirs[] = {
+        ba("/opt/etc/ssl/certs/"), // Entware
         ba("/etc/ssl/certs/"), // (K)ubuntu, OpenSUSE, Mandriva ...
         ba("/usr/lib/ssl/certs/"), // Gentoo, Mandrake
         ba("/usr/share/ssl/"), // Red Hat pre-2004, SuSE
