--- a/src/corelib/global/qsysinfo.cpp
+++ b/src/corelib/global/qsysinfo.cpp
@@ -310,7 +310,8 @@ static bool readOsRelease(QUnixOSVersion
     // Applications should check for the former, and exclusively use its data
     // if it exists, and only fall back to /usr/lib/os-release if it is
     // missing.
-    return readEtcFile(v, "/etc/os-release", id, versionId, prettyName) ||
+    return readEtcFile(v, "/opt/etc/os-release", id, versionId, prettyName) ||
+            readEtcFile(v, "/etc/os-release", id, versionId, prettyName) ||
             readEtcFile(v, "/usr/lib/os-release", id, versionId, prettyName);
 }
 
@@ -323,7 +324,7 @@ static bool readEtcLsbRelease(QUnixOSVer
         // so try /etc/<lowercasename>-release
 
         // we're still avoiding QFile here
-        QByteArray distrorelease = "/etc/" + v.productType.toLatin1().toLower() + "-release";
+        QByteArray distrorelease = "/opt/etc/" + v.productType.toLatin1().toLower() + "-release";
         int fd = qt_safe_open(distrorelease, O_RDONLY);
         if (fd != -1) {
             QT_STATBUF sbuf;
--- a/src/corelib/io/qstandardpaths_unix.cpp
+++ b/src/corelib/io/qstandardpaths_unix.cpp
@@ -395,7 +395,7 @@ static QStringList xdgDataDirs()
 
     QStringList dirs = dirsList(xdgDataDirsEnv);
     if (dirs.isEmpty())
-        dirs = QStringList{u"/usr/local/share"_s, u"/usr/share"_s};
+        dirs = QStringList{u"/opt/usr/share"_s, u"/opt/share"_s, u"/usr/local/share"_s, u"/usr/share"_s};
 
     return dirs;
 }
@@ -407,7 +407,7 @@ static QStringList xdgConfigDirs()
 
     QStringList dirs = dirsList(xdgConfigDirs);
     if (dirs.isEmpty())
-        dirs.push_back(u"/etc/xdg"_s);
+        dirs.push_back(u"/opt/etc/xdg"_s);
 
     return dirs;
 }
--- a/src/corelib/io/qstorageinfo_p.h
+++ b/src/corelib/io/qstorageinfo_p.h
@@ -152,8 +152,8 @@ bool QStorageInfoPrivate::shouldIncludeF
     if (isParentOf("/dev"_L1, mountDir)
         || isParentOf("/proc"_L1, mountDir)
         || isParentOf("/sys"_L1, mountDir)
-        || isParentOf("/var/run"_L1, mountDir)
-        || isParentOf("/var/lock"_L1, mountDir)) {
+        || isParentOf("/opt/var/run"_L1, mountDir)
+        || isParentOf("/opt/var/lock"_L1, mountDir)) {
         return false;
     }
 
--- a/src/corelib/time/qtimezoneprivate_tz.cpp
+++ b/src/corelib/time/qtimezoneprivate_tz.cpp
@@ -64,14 +64,14 @@ static bool openZoneInfo(const QString &
             return true;
     }
     // Try modern system path first:
-    constexpr auto zoneShare = "/usr/share/zoneinfo/"_L1;
+    constexpr auto zoneShare = "/opt/share/zoneinfo/"_L1;
     if (tzdir != zoneShare && tzdir != zoneShare.chopped(1)) {
         file->setFileName(zoneShare + name);
         if (file->open(QIODevice::ReadOnly))
             return true;
     }
     // Fall back to legacy system path:
-    constexpr auto zoneLib = "/usr/lib/zoneinfo/"_L1;
+    constexpr auto zoneLib = "/opt/lib/zoneinfo/"_L1;
     if (tzdir != zoneLib && tzdir != zoneLib.chopped(1)) {
         file->setFileName(zoneShare + name);
         if (file->open(QIODevice::ReadOnly))
@@ -792,7 +792,7 @@ QTzTimeZoneCacheEntry QTzTimeZoneCache::
     QFile tzif;
     if (ianaId.isEmpty()) {
         // Open system tz
-        tzif.setFileName(QStringLiteral("/etc/localtime"));
+        tzif.setFileName(QStringLiteral("/opt/etc/localtime"));
         if (!tzif.open(QIODevice::ReadOnly))
             return ret;
     } else if (!openZoneInfo(QString::fromLocal8Bit(ianaId), &tzif)) {
@@ -1310,9 +1310,9 @@ public:
            The alternative would be to use a file-system watcher, but they are a
            scarce resource.
          */
-        const StatIdent local = identify("/etc/localtime");
-        const StatIdent tz = identify("/etc/TZ");
-        const StatIdent timezone = identify("/etc/timezone");
+        const StatIdent local = identify("/opt/etc/localtime");
+        const StatIdent tz = identify("/opt/etc/TZ");
+        const StatIdent timezone = identify("/opt/etc/timezone");
         if (!m_name.isEmpty() && m_last.isValid()
             && (m_last == local || m_last == tz || m_last == timezone)) {
             return m_name;
@@ -1325,14 +1325,14 @@ public:
         }
 
         // Some systems (e.g. uClibc) have a default value for $TZ in /etc/TZ:
-        m_name = etcContent(QStringLiteral("/etc/TZ"));
+        m_name = etcContent(QStringLiteral("/opt/etc/TZ"));
         if (!m_name.isEmpty()) {
             m_last = tz;
             return m_name;
         }
 
         // Gentoo still (2020, QTBUG-87326) uses this:
-        m_name = etcContent(QStringLiteral("/etc/timezone"));
+        m_name = etcContent(QStringLiteral("/opt/etc/timezone"));
         m_last = m_name.isEmpty() ? StatIdent() : timezone;
         return m_name;
     }
@@ -1363,7 +1363,7 @@ private:
         // name from the path
         const QString tzdir = qEnvironmentVariable("TZDIR");
         constexpr auto zoneinfo = "/zoneinfo/"_L1;
-        QString path = QStringLiteral("/etc/localtime");
+        QString path = QStringLiteral("/opt/etc/localtime");
         long iteration = getSymloopMax();
         // Symlink may point to another symlink etc. before being under zoneinfo/
         // We stop on the first path under /zoneinfo/, even if it is itself a
@@ -1432,7 +1432,7 @@ QByteArray QTzTimeZonePrivate::staticSys
     // to be a "default timezone", in which case it will be read
     // by one of the blocks below, so unset it here so it is not
     // considered as a valid/found ianaId
-    if (ianaId == ":/etc/localtime")
+    if (ianaId == ":/opt/etc/localtime")
         ianaId.clear();
     else if (ianaId.startsWith(':'))
         ianaId = ianaId.sliced(1);
--- a/src/plugins/tls/openssl/qtlsbackend_openssl.cpp
+++ b/src/plugins/tls/openssl/qtlsbackend_openssl.cpp
@@ -389,6 +389,7 @@ QList<QSslCertificate> systemCaCertifica
     {
         const QList<QByteArray> directories = QSslSocketPrivate::unixRootCertDirectories();
         QSet<QString> certFiles = {
+            QStringLiteral("/opt/etc/pki/tls/certs/ca-bundle.crt"), // Entware
             QStringLiteral("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"), // Red Hat 2013+
             QStringLiteral("/etc/pki/tls/certs/ca-bundle.crt"), // Red Hat older, Mandriva
             QStringLiteral("/usr/local/share/certs/ca-root-nss.crt") // FreeBSD's ca_root_nss
