--- a/accel-pppd/cli/telnet.c
+++ b/accel-pppd/cli/telnet.c
@@ -69,7 +69,7 @@ static uint8_t *recv_buf;
 static uint8_t *temp_buf;
 
 static int conf_history_len = 100;
-static const char *conf_history_file = "/var/lib/accel-ppp/history";
+static const char *conf_history_file = "/opt/var/lib/accel-ppp/history";
 static int conf_verbose;
 
 static LIST_HEAD(history);
--- a/accel-pppd/logs/log_file.c
+++ b/accel-pppd/logs/log_file.c
@@ -531,7 +531,7 @@ static void ev_ctrl_started(struct ap_se
 
 		lpd->tmp = temp_seq++;
 		strcpy(fname, conf_per_session_dir);
-		strcat(fname, "/tmp");
+		strcat(fname, "/opt/tmp");
 		sprintf(fname + strlen(fname), "%lu", lpd->tmp);
 
 		if (log_file_open(&lpd->lf, fname)) {
@@ -581,7 +581,7 @@ static void ev_ctrl_finished(struct ap_s
 			fname = _malloc(PATH_MAX);
 			if (fname) {
 				strcpy(fname, conf_per_session_dir);
-				strcat(fname, "/tmp");
+				strcat(fname, "/opt/tmp");
 				sprintf(fname + strlen(fname), "%lu", lpd->tmp);
 				if (unlink(fname))
 					log_emerg("log_file: unlink '%s': %s\n", fname, strerror(errno));
@@ -616,7 +616,7 @@ static void ev_ses_starting(struct ap_se
 	}
 
 	strcpy(fname1, conf_per_session_dir);
-	strcat(fname1, "/tmp");
+	strcat(fname1, "/opt/tmp");
 	sprintf(fname1 + strlen(fname1), "%lu", lpd->tmp);
 
 	strcpy(fname2, conf_per_session_dir);
--- a/accel-pppd/net.c
+++ b/accel-pppd/net.c
@@ -447,7 +447,7 @@ static void __init init()
 	if (opt)
 		conf_netns_run_dir = opt;
 	else
-		conf_netns_run_dir = "/var/run/netns";
+		conf_netns_run_dir = "/opt/var/run/netns";
 
 	def_net = net = alloc_net(NULL);
 }
--- a/accel-pppd/session.c
+++ b/accel-pppd/session.c
@@ -540,7 +540,7 @@ static void load_config(void)
 
 	conf_seq_file = conf_get_opt("common", "seq-file");
 	if (!conf_seq_file)
-		conf_seq_file = "/var/lib/accel-ppp/seq";
+		conf_seq_file = "/opt/var/lib/accel-ppp/seq";
 
 	opt = conf_get_opt("common", "max-sessions");
 	if (opt)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,7 +84,7 @@ if (NOT BUILD_DRIVER_ONLY)
 	add_subdirectory(accel-pppd)
 	add_subdirectory(accel-cmd)
 
-	install(DIRECTORY DESTINATION /var/log/accel-ppp)
+	install(DIRECTORY DESTINATION /opt/var/log/accel-ppp)
 endif (NOT BUILD_DRIVER_ONLY)
 
 if (BUILD_PPTP_DRIVER)
